src/lib/CurrencyInput.svelte

```diff
@@ -47,42 +47,56 @@
+		if (event) {
+			// Don't format if the user is typing a `currencyDecimal` point
+			if (event.key === currencyDecimal) return;
 
-		// Pressing `.` when the decimal point is `,` gets replaced with `,`
-		if (isDecimalComma && event.key === '.')
-			formattedValue = formattedValue.replace(/\.([^.]*)$/, currencyDecimal + '$1'); // Only replace the last occurence
+			// Pressing `.` when the decimal point is `,` gets replaced with `,`
+			if (isDecimalComma && event.key === '.')
+				formattedValue = formattedValue.replace(/\.([^.]*)$/, currencyDecimal + '$1'); // Only replace the last occurence
 
-		// Pressing `,` when the decimal point is `.` gets replaced with `.`
-		if (!isDecimalComma && event.key === ',')
-			formattedValue = formattedValue.replace(/\,([^,]*)$/, currencyDecimal + '$1'); // Only replace the last occurence
+			// Pressing `,` when the decimal point is `.` gets replaced with `.`
+			if (!isDecimalComma && event.key === ',')
+				formattedValue = formattedValue.replace(/\,([^,]*)$/, currencyDecimal + '$1'); // Only replace the last occurence
 
-		// Don't format if `formattedValue` is ['$', '-$', "-"]
-		const ignoreSymbols = [currencySymbol, `-${currencySymbol}`, '-'];
-		const strippedUnformattedValue = formattedValue.replace(' ', '');
-		if (ignoreSymbols.includes(strippedUnformattedValue)) return;
+			// Don't format if `formattedValue` is ['$', '-$', "-"]
+			const ignoreSymbols = [currencySymbol, `-${currencySymbol}`, '-'];
+			const strippedUnformattedValue = formattedValue.replace(' ', '');
+			if (ignoreSymbols.includes(strippedUnformattedValue)) return;
 
-		// Set the starting caret positions
-		inputTarget = event.target as HTMLInputElement;
+			// Set the starting caret positions
+			inputTarget = event.target as HTMLInputElement;
 
+			// Reverse the value when minus is pressed
+			if (isNegativeAllowed && event.key === '-') value = value * -1;
+
+		}
+		
 		// Remove all characters that arent: numbers, commas, periods (or minus signs if `isNegativeAllowed`)
 		let unformattedValue = isNegativeAllowed
 			? formattedValue.replace(/[^0-9,.-]/g, '')
 			: formattedValue.replace(/[^0-9,.]/g, '');
 
-		// Reverse the value when minus is pressed
-		if (isNegativeAllowed && event.key === '-') value = value * -1;
-
 		// Finally set the value
 		if (Number.isNaN(parseFloat(unformattedValue))) {
 			value = 0;
 		} else {
 			// The order of the following operations is *critical*
 			unformattedValue = unformattedValue.replace(isDecimalComma ? /\./g : /\,/g, ''); // Remove all group symbols
 			if (isDecimalComma) unformattedValue = unformattedValue.replace(',', '.'); // If the decimal point is a comma, replace it with a period
+			// If the zero-key has been pressed
+			// and if the current `value` is the same as the `value` before the key-press
+			// formatting may need to be done (Issue #30)
+			let prevValue = value;
 			value = parseFloat(unformattedValue);
+			if (event && prevValue == value) {
```

Comment:
```suggestion

			// If the zero-key has been pressed
			// and if the current `value` is the same as the `value` before the key-press
			// formatting may need to be done (Issue #30)
			const previousValue = value;
			value = parseFloat(unformattedValue);

			if (event && previousValue === value) {
```

- Added extra line before the comment and before the `if` statement.
- Personal preference not using abbreviated names: `prevValue` > `previousValue`
- Use `const` if we are not going to reassign the value: `let prevValue` > `const previousValue`
- Match the value and the type: `prevValue == value` > `previousValue === value`

Reply:
Sure, will make the all required changes by tomorrow.